src/v/kafka/server/tests/usage_manager_test.cc

```diff
@@ -0,0 +1,198 @@
+    const auto window_width = 10s;
+    const auto disk_write_interval = 1s;
+
+    auto usage_fiber = std::make_unique<test_accounting_fiber>(
+      *kvstore, num_windows, window_width, disk_write_interval);
+    usage_fiber->start().get();
+
+    /// Create expected data set
+    std::vector<kafka::usage> data;
+    for (auto i = 0; i < num_windows; ++i) {
+        const uint64_t sent = i * 100;
+        const uint64_t recv = i * 200;
+        data.emplace_back(
+          kafka::usage{.bytes_sent = sent, .bytes_received = recv});
+    }
+
+    /// Publish some data as the windows move across in time, assert observed is
+    /// as expected
+    for (const auto& e : data) {
+        usage_fiber->add_bytes(e.bytes_sent, e.bytes_received);
+        usage_fiber->advance_clock(window_width).get();
+        vlog(
+          af_logger.info,
+          "Clock advanced: by {}s, data: {}",
+          window_width.count(),
+          print_window_data(usage_fiber->get_usage_stats().get()));
+    }
+    auto result = strip_window_data(usage_fiber->get_usage_stats().get());
+    BOOST_CHECK_EQUAL(result, data);
+
+    /// Add to open window
+    usage_fiber->add_bytes(10, 10);
+    usage_fiber->advance_clock(2s).get();
+    usage_fiber->add_bytes(10, 10);
+    usage_fiber->advance_clock(2s).get();
+    const auto open_windows = usage_fiber->get_usage_stats().get();
+    vlog(
+      af_logger.info,
+      "Clock advanced 4s, data: {}",
+      print_window_data(open_windows));
+    const auto open_window = open_windows[0];
+    BOOST_CHECK_EQUAL(open_window.u.bytes_sent, 20);
+    BOOST_CHECK_EQUAL(open_window.u.bytes_received, 20);
+
+    /// Shut it down, note clean shutdowns persist data to kvstore
+    usage_fiber->stop().get();
+    usage_fiber = nullptr;
+
+    /// Restart the instance, and verify expected behavior
+    usage_fiber = std::make_unique<test_accounting_fiber>(
+      *kvstore, num_windows, window_width, disk_write_interval);
+    usage_fiber->start().get();
+
+    // Ensure open window is consistent
+    result = strip_window_data(usage_fiber->get_usage_stats().get());
+    const auto& new_open_window = result[0];
+    BOOST_CHECK_EQUAL(new_open_window.bytes_sent, 20);
+    BOOST_CHECK_EQUAL(new_open_window.bytes_received, 20);
+    BOOST_CHECK_EQUAL(result, data);
+
+    usage_fiber->stop().get();
+    kvstore->stop().get();
+}
+
+SEASTAR_THREAD_TEST_CASE(test_round_to_interval_method) {
+    using namespace std::chrono_literals;
+    /// Aug 3rd, 2023 4PM GMT
+    const auto ts = ss::lowres_system_clock::time_point(
+      std::chrono::seconds(1691078400));
+    /// 15min, 30min, 1hr, 2hrs
+    const auto intervals = std::to_array({900s, 1800s, 3600s, 7200s});
+
+    for (const auto& ival : intervals) {
+        /// Underneath the 2min threshold the method returns the input rounded
+        BOOST_CHECK(ts == kafka::detail::round_to_interval(ival, ts));
+        BOOST_CHECK(ts == kafka::detail::round_to_interval(ival, ts + 10s));
```

Comment:
in a live `system clock::now` is passed to `round_to_interval`, so any timestamp may be passed in. however, in this test `1691078400` happens to be divisible by each of the numbers in the `intervals` set. if we change the timestamp to be `1691078400 + 1` the test fails, so it doesn't seem to be testing an expected property of `round_to_interval`.

part of the confusion here is that it's not clear what the semantics are expected to be of `round_to_interval`.